mm2src/lp_ordermatch.rs

```diff
@@ -1905,8 +2040,7 @@ pub async fn lp_auto_buy(
         Some("sell") => TakerAction::Sell,
         _ => return ERR!("Auto buy must be called only from buy/sell RPC methods"),
     };
-    let topic = orderbook_topic(&input.base, &input.rel);
-    subscribe_to_topic(ctx, topic.clone()).await;
+    try_s!(subscribe_to_orderbook_topic(&ctx, &input.base, &input.rel).await);
```

Comment:
`subscribe_to_orderbook_topic` also requests the orderbook from relayers which might be an unwanted side effect during `buy/sell/setprice` calls. When we place an order we don't mandatory need the corresponding pair orderbook so we can just subscribe here.

Reply:
It could be a reason for orders losing:

- call `byu/sell/setprice`, MM2 subscribes to a topic and doesn't request any order
- call `orderbook`, in the OrdermatchContext::topic_subscribed_to contains corresponding topic, and we think that the orderbook for the given coins is empty

Even if we will check if the orderbook is empty, we could receive an order between `byu/sell/setprice` and `orderbook` calls

Reply:
True, maybe we can track the state whether orderbook was requested with timestamp? And `orderbook` RPC logic might be like this:
1. If orderbook has been requested from relays already return the local known orderbook.
1. If orderbook has not been requested, but we are subscribed to topic for long enough (a bit more than order keep alive period) return the local known orderbook.
1. If orderbook has not been requested and we're subscribed recently then request the orders from relayers.

Reply:
Sounds good. Ok, lets do this :+1: